### PR TITLE
enforce reg-sub :<- syntax

### PR DESCRIPTION
**Pull request by Sohalt**
This implements #326

### CODE REVIEW

File: src/re_frame/subs.cljc [hunk 1]
```diff
@@ -183,14 +183,18 @@
                              f)
 
                          ;; one sugar pair
-                         2 (fn inp-fn
-                             ([_] (subscribe (second input-args)))
-                             ([_ _] (subscribe (second input-args))))
+                         2 (let [[label vec] input-args]
+                             (when-not (= :<- )
```
Comment (mike-thompson-day8) [inline src/re_frame/subs.cljc]: Shouldn't this be:   `(when-not (= :-< label))`

**Comment by Sohalt** [comment]
Yes, obviously. I don't know how I did miss that. Thanks for catching it.

**Comment by Sohalt** [comment]
I amended the fix. Does the rest look ok to you?

**Comment by danielcompton** [comment]
This looks pretty good. I've got some style tweaks and I might add some tests for this behaviour, but it looks sound. Thanks!

File: src/re_frame/subs.cljc [hunk 2]
```diff
@@ -183,14 +183,18 @@
                              f)
 
                          ;; one sugar pair
-                         2 (fn inp-fn
-                             ([_] (subscribe (second input-args)))
-                             ([_ _] (subscribe (second input-args))))
+                         2 (let [[label vec] input-args]
```
Comment (danielcompton) [inline src/re_frame/subs.cljc]: `vec` should be `query-v` to not shadow `cljs.core/vec`

Comment (danielcompton) [inline src/re_frame/subs.cljc]: I'm not that keen on `label` here, but I'm struggling to think of a different name. Perhaps `directive`?

Comment (mike-thompson-day8) [inline src/re_frame/subs.cljc]: beacon? marker? glue? sentinel?

Comment (mike-thompson-day8) [inline src/re_frame/subs.cljc]: delimiter?

Comment (mainej) [inline src/re_frame/subs.cljc]: sigil? token?

Comment (Sohalt) [inline src/re_frame/subs.cljc]: Any consensus? Out of the suggestions I personally like `marker` the most.

**Comment by danielcompton** [comment]
Thanks @Sohalt, I've merged it with `marker` for now.

**Pull request merged**